04-Games/src/js/games/FallingStones.js

```diff
@@ -38,18 +39,21 @@ export class FallingStones extends GameTemplate {
     update(ctx) {
         this.player.update(ctx);
         this.player.borderCollision(ctx);
-        for (let i = 0; i < this.bullets.length; ++i) {
+        for (let i = this.bullets.length -1 ; i >= 0 ; i--) {
             this.bullets[i].update(ctx); //update coordinates of all projectiles
-            for (let m = 0; m < this.stones.length; ++m) {
+            for (let m = this.stones.length -1; m >= 0; m--) {
                 if (this.bullets[i].y >= this.stones[m].y && this.bullets[i].y <= this.stones[m].y + 100){
                     if (this.bullets[i].x >= this.stones[m].x && this.bullets[i].x <= this.stones[m].x + 50) {
                         this.bullets.splice(i, 1), //removes projectil and stone by contact
                         this.stones.splice(m, 1);
                         this.points++;
+                        break;
```

Comment:
Dir scheint schon grundlegend klar zu sein dass du nach der ersten kugel die kollidiert die anderen diesen frame nicht mehr anschauen musst. 
Mit diesem Fakt kannst du aber auch eine for schleife rausnehmen, du musst nur eine Kugel prüfen pro frame. (solange kugeln und steine konstante geschwindigkeiten haben)

Reply:
Bisher hab ich nicht herausgefunden, welche Bedingung ich für die Kugeln dann aufstellen müsste, dass nur die eine ausschlaggebende Kugel überprüft wird.

Reply:
du kannst einfach nur die erste Kugel prüfen. durch die pause zwischen den Schüssen ist es unwahrscheinlich dass die erste noch im bild ist während die zweite gerade noch kollidiert. Dadurch ist die zweite die erste und prüft korrekt auf Kollision. Es ist ja nicht entscheidend dass du die kollision sofort erkennst. Solange du sie erkennst bevor sie endet reichts in dem Spiel ja.